pep/app/authenticators/credentials.ts

```diff
@@ -0,0 +1,164 @@
+import { inject as service } from '@ember/service';
+import { isEmpty } from '@ember/utils';
+import { resolve, reject } from 'rsvp';
+import BaseAuthenticator from 'ember-simple-auth/authenticators/base';
+
+import AjaxService from 'pep/services/ajax';
+import ENV from 'pep/config/environment';
+import { PepSecureAuthenticatedData } from 'pep/api';
+import { serializeQueryParams } from 'pep/utils/url';
+import PepSessionService from 'pep/services/pep-session';
+import { run } from '@ember/runloop';
+import { EmberRunTimer } from '@ember/runloop/types';
+import FastbootService from 'ember-cli-fastboot/services/fastboot';
+
+export enum SessionType {
+    CREDENTIALS = 'credentials',
+    IP = 'ip'
+}
+
+export default class CredentialsAuthenticator extends BaseAuthenticator {
+    @service ajax!: AjaxService;
+    @service('pep-session') session!: PepSessionService;
+    @service fastboot!: FastbootService;
+
+    authenticationHeaders = {
+        'Content-Type': 'application/json'
+    };
+    private _invalidateTimeout?: EmberRunTimer;
+
+    /**
+     * Calculate expiration time - copied from ember simple auth
+     *
+     * @param {number} expiresIn
+     * @returns
+     * @memberof CredentialsAuthenticator
+     */
+    _absolutizeExpirationTime(expiresIn: number) {
+        if (!isEmpty(expiresIn)) {
+            return new Date(new Date().getTime() + expiresIn * 1000).getTime();
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Calculate the expires at value and add it to the response so it gets saved in the cookie
+     *
+     * @param {PepSecureAuthenticatedData} response
+     * @returns
+     * @memberof CredentialsAuthenticator
+     */
+    setupExpiresAt(response: PepSecureAuthenticatedData) {
+        const expiresAt = this._absolutizeExpirationTime(response.SessionExpires);
+        if (expiresAt && !isEmpty(expiresAt)) {
+            Object.assign(response, { expiresAt: expiresAt });
+        }
+        return response;
+    }
+
+    /**
+     * Authenticates and logs the user in as well as schedules a session expiration based on the
+     * SessionExpires number that is sent back in the result
+     * @param  {String} username
+     * @param {String} password
+     */
+    async authenticate(username: string, password: string) {
+        try {
+            const sessionData = this.session.getUnauthenticatedSession();
+            const response = await this.ajax.request<PepSecureAuthenticatedData>(`${ENV.authBaseUrl}/Authenticate`, {
+                method: 'POST',
+                headers: this.authenticationHeaders,
+                body: this.ajax.stringifyData({
+                    grant_type: 'password',
+                    SessionId: sessionData?.SessionId,
```

Comment:
@bakerac4 missed this in my last pass. but looks like `grant_type` no longer exists in the defined request body for this endpoint in the swagger docs (i think that was a holdover from the original PEP API login).

Also, it appears `SessionId` is a query param, not a body field, unless the docs are wrong on that.